client/src/providers/favorites.tsx

```diff
@@ -0,0 +1,71 @@
+import _ from 'lodash';
+import React, {useState, useEffect, useContext} from 'react';
+import {AsyncStorage} from 'react-native';
+import * as Types from '../types/graphql';
+import {messageBox} from '../services';
+
+export type Favorite = Types.Poi | Types.Route;
+
+type FavoritesContextType = {
+    favorites: Favorite[],
+    isFavorite: (item: Types.Poi) => boolean,
+    addFavorite: (item: Types.Poi) => void,
+    removeFavorite: (item: Types.Poi) => void,
+};
+
+type FavoritesProviderProps = {
+    children: React.ReactNode,
+};
+
+const favoritesStorageKey = 'favorites';
+
+const defaultValue: FavoritesContextType = {
+    favorites: [],
+    isFavorite: () => false,
+    addFavorite: _.noop,
+    removeFavorite: _.noop,
+};
+
+const FavoritesContext = React.createContext(defaultValue);
+
+export const FavoritesProvider: React.FC<FavoritesProviderProps> = ({children}) => {
```

Comment:
btw, it has types for children by default

Reply:
noted